.teamcity/src/subprojects/build/core/ProjectCore.kt

```diff
@@ -35,37 +42,41 @@ object ProjectCore : Project({
         defaultTimeouts()
     }
 
-    val osJVMCombos = operatingSystems.flatMap { os ->
+    val OsJdk = operatingSystems.flatMap { os ->
         jdkVersions.map { jdk -> OSJDKEntry(os, jdk) }
     }
 
-    val osJVMBuilds = osJVMCombos.map(::CoreBuild)
+    val osJdkBuilds = OsJdk.map(::CoreBuild)
     val nativeBuilds = operatingSystems.map(::NativeBuild)
     val javaScriptBuilds = javaScriptEngines.map(::JavaScriptBuild)
     val stressTestBuilds = stressTests.map(::StressTestBuild)
 
-    val allBuilds = osJVMBuilds.plus(nativeBuilds).plus(javaScriptBuilds).plus(stressTestBuilds)
+    val allBuilds = osJdkBuilds.plus(nativeBuilds).plus(javaScriptBuilds).plus(stressTestBuilds)
 
     allBuilds.forEach(::buildType)
 
     buildType {
-        id("KtorCore_All")
-        name = "Build All Core"
-        type = BuildTypeSettings.Type.COMPOSITE
+        createCompositeBuild("KtorCore_All", "Build All Core", VCSCore, allBuilds)
+    }
+})
 
-        vcs {
-            root(VCSCore)
-        }
+fun BuildType.createCompositeBuild(buildId: String, buildName: String, vcsRoot: VcsRoot, builds: List<BuildType>) {
```

Comment:
Too many parameters in the function. I think `buildId` and `buildName` could be as one parameter. Also, I suggest renaming `builds` to `dependentBuilds` or something similar for clarity.